src/CBT.NuGet/Internal/AggregatePackage.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace CBT.NuGet.Internal
+{
+    internal sealed class AggregatePackage
+    {
+        public enum AggregateOperation
+        {
+            Add = '*',
+            Remove = '!',
+        }
+
+        internal struct PackageOperation
+        {
+            internal AggregateOperation Operation { get; set; }
+
+            internal string Folder { get; set; }
+        }
+
+        public AggregatePackage(string outPropertyId, ICollection<PackageOperation> packagesToAggregate, string destinationRoot)
+        {
+            OutPropertyId = outPropertyId;
+            PackagesToAggregate = packagesToAggregate;
+            OutPropertyValue = Path.Combine(destinationRoot, $"{outPropertyId}.{GetOutputPropertyHash()}");
+        }
+
+        public string OutPropertyId { get; private set; }
+
+        public string OutPropertyValue { get; private set; }
+
+        public ICollection<PackageOperation> PackagesToAggregate { get; private set; }
+
+        private string GetOutputPropertyHash()
+        {
+            StringBuilder valueToHash = new StringBuilder();
+            foreach (var pkg in PackagesToAggregate)
+            {
+                valueToHash.Append(pkg.Folder.ToLower());
+                // Include last write time on all files to ensure that if the source folder was updated a new aggregate package will be created.
+                // if a file was removed then it will not be in the value to hash so the hash will still change to trigger a new aggregation as desired.
+                // We don't care about directories in aggregation since removing directories is not possible given current implementation.
+                // This is in the scenario where the user aggregates against a nuget package and a checked in folder under source control that contains a config file or whatever.
+                // Concern on aggregating very large folders as this is disk IO.
+                foreach (var f in Directory.GetFiles(pkg.Folder, "*.*", SearchOption.AllDirectories))
```

Comment:
Actually I have an idea. I will send an update.  Basically we only care about non immutable folders.  So we can pass a list of immutable roots and that can be a ; delimited list of paths and by default that can be the nuget root.  This would resolve the issue in the 99% case.  In the 1% case the user has the escape hatch to say their immutable folder is really a non-immutable one.